src/server/ApiDispatchers.cpp

```diff
@@ -191,30 +192,13 @@ constexpr T saturate(auto val)
     size_t cbWritten;
     LOG_IF_FAILED(SizeTMult(outEvents.size(), sizeof(INPUT_RECORD), &cbWritten));
 
-    if (nullptr != waiter.get())
+    if (waiter)
     {
-        // In some circumstances, the read may have told us to wait because it didn't have data,
-        // but the client explicitly asked us to return immediate. In that case, we'll convert the
-        // wait request into a "0 bytes found, OK".
-
-        if (fIsWaitAllowed)
-        {
-            hr = ConsoleWaitQueue::s_CreateWait(m, waiter.release());
-            if (SUCCEEDED(hr))
-            {
-                *pbReplyPending = TRUE;
-                hr = CONSOLE_STATUS_WAIT;
-            }
-        }

```

Comment:
Makes no sense IMO to check this here if we can just pass down `fIsWaitAllowed` and avoid a waiter from being created in the first place.